test/e2e/config/ironic.yaml

```diff
@@ -2,6 +2,8 @@ images:
 # Use locally built e2e images
 - name: quay.io/metal3-io/baremetal-operator:e2e
   loadBehavior: tryLoad
+# - name: quay.io/metal3-io/ironic:local
```

Comment:
Why have this added but commented out?

Reply:
Oh I forgot about it!
@Rozzii you added the option to use a locally built ironic image recently. This is basically all that is required for it as far as I understand. If you uncomment this, the image is loaded. The rest is handled by changing the manifests directly or by pointing to another kustomization to deploy below.
Is this acceptable?

Previous process:
- Set `LOAD_LOCAL_IRONIC=true` and build the image
- Change e2e config to deploy ironic from [e2e-local-ironic](https://github.com/metal3-io/baremetal-operator/tree/main/ironic-deployment/overlays/e2e-local-ironic)

Process after this PR:
- Build the image and comment out these lines in e2e config
- Change e2e config to deploy ironic from e2e-local-ironic

Reply:
If we move to kind then the image transfer is not needed anymore  as kind runs on the host VM so in that context I am fine with the changes.
 
I am fine with doing it via the config file although I prefer env variable based config because I am a bit worried that it is easy to accidentally commit changes to ironic.yaml.